.github/workflows/checks.yml

```diff
@@ -197,6 +200,42 @@ jobs:
       - name: Install devnet
         run: ./starknet_py/tests/install_devnet.sh
 
+      # ====================== SETUP LEDGER SPECULOS ====================== #
+
+      - name: Pull speculos image
+        run: docker pull ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools
+
+
+      - name: Clone LedgerHQ Starknet app repository
+        run: git clone https://github.com/LedgerHQ/app-starknet.git
+
+      - name: Build the app inside Docker container
+        uses: addnab/docker-run-action@v3
+        with:
+          image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools
+          options: --rm -v ${{ github.workspace }}:/apps
+          run: |
+            cd /apps/app-starknet
+            cargo clean
+            cargo ledger build nanox
+            
+      - name: Start Speculos emulator container
+        uses: addnab/docker-run-action@v3
+        with:
+          image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-dev-tools
+          options: --rm -d --name speculos-emulator -v ${{ github.workspace }}:/apps --publish 5001:5001 --publish 9999:9999
+          run: speculos -m nanox --apdu-port 9999 --api-port 5001 --display headless /apps/app-starknet/target/nanox/release/starknet --automation '{"version":1,"rules":[{"text":"Confirm Hash to sign","conditions":[],"actions":[["button",2,true],["button",2,false]]},{"regexp":".*Hash \\(.*\\)","conditions":[],"actions":[["button",2,true],["button",2,false]]},{"text":"Reject","conditions":[],"actions":[["button",2,true],["button",2,false]]},{"text":"Approve","conditions":[],"actions":[["button",1,true],["button",2,true],["button",1,false],["button",2,false]]}]}'
```

Comment:
note: This will be moved to separate file. For now I've put the automation json here, in order to check if it works properly.